autotest/gcore/rasterio.py

```diff
@@ -845,6 +845,27 @@ def test_rasterio_16():
 
 ###############################################################################
 
+def test_rasterio_nodata():
+
+        ndv = 123
+        btype = [ gdal.GDT_Byte, gdal.GDT_Int16, gdal.GDT_Int32, gdal.GDT_Float32, gdal.GDT_Float64 ]
+
+        ### create a MEM dataset
+        for src_type in btype:
+                mem_ds = gdal.GetDriverByName('MEM').Create('', 10, 9, 1, src_type)
+                mem_ds.GetRasterBand(1).SetNoDataValue(ndv)
+                mem_ds.GetRasterBand(1).Fill(ndv)
+
+                for dst_type in btype:
+                        if ( dst_type > src_type ):
+
+                                ### read to a buffer of a wider type (and resample)
+                                data = mem_ds.GetRasterBand(1).ReadAsArray(0, 0, 10, 9, 4, 3, resample_alg=gdal.GRIORA_Bilinear, buf_type=dst_type)
```

Comment:
not all builds in the continuous integration includes gdalnumeric / numpy support, and ReadAsArray() requires it. So either add at the beginning of this test a detection of gdalnumeric (see for example test_rasterio_13), or use lower level ReadRaster() with struct.unpack (other tests in that file do that)

Reply:
added the skip code...
not sure how to quickly map it into using struct.unpack